trace/trace.go

```diff
@@ -16,17 +14,30 @@ type Trace struct {
 	ProbabilitySample *float64
 }
 
-func newTrace(defaultProbabilitySample float64) Trace {
+func newTrace() Trace {
 	return Trace{
 		ID:                NewTraceID(),
 		ProbabilitySample: &defaultProbabilitySample,
 	}
 }
 
-func createTraceIfEmpty(ctx context.Context, t *Trace, defaultProbabilitySample float64) context.Context {
-	if t == nil || IDIsEmpty(t.ID) || t.ProbabilitySample == nil {
-		*t = newTrace(defaultProbabilitySample)
+func ensureTraceNotEmpty(t *Trace, shouldLogIfIsSomethingEmpty bool) {
```

Comment:
This flag  `shouldLogIfIsSomethingEmpty` (I'm assuming it's used for debugging) could be a global variable meant to be set once (in main, for example) or you could just log in debug mode. Or we can always log it if this is really important.